scripts/dev/custom_check.sh

```diff
@@ -16,5 +16,6 @@ python3 "$REPO_ROOT"/scripts/dev/check_for_bom_in_idfs.py || EXIT_STATUS=$?
 python3 "$REPO_ROOT"/scripts/dev/verify_cmake_dirs.py || EXIT_STATUS=$?
 python3 "$REPO_ROOT"/scripts/dev/find_included_cc_files.py || EXIT_STATUS=$?
 python3 "$REPO_ROOT"/scripts/dev/analyze_state.py || EXIT_STATUS=$?
+python3 "$REPO_ROOT"/scripts/dev/check_for_c_style_comments.py || EXIT_STATUS=$?
```

Comment:
Added the new script to the custom_check so it is executed by GHA on every commit.